pkgs/build-support/docker/default.nix

```diff
@@ -994,4 +996,177 @@ rec {
+    , # Same as `command`, but runs the command in a non-interactive shell instead. See `--run` in `man nix-shell`
+      run ? null
+    }:
+      assert lib.assertMsg (! (drv.drvAttrs.__structuredAttrs or false))
+        "buildDerivationImage: Does not work with the derivation ${drv.name} because it uses __structuredAttrs";
+      assert lib.assertMsg (command == null || run == null)
+        "buildDerivationImage: Can't specify both command and run";
+      let
+
+        # A binary that calls the command to build the derivation
+        builder = writeShellScriptBin "buildDerivation" ''
+          exec ${lib.escapeShellArg (stringValue drv.drvAttrs.builder)} ${lib.escapeShellArgs (map stringValue drv.drvAttrs.args)}
+        '';
+
+        # https://github.com/NixOS/nix/blob/2.8.0/src/nix-build/nix-build.cc#L493-L526
+        rcfile = writeText "nix-shell-rc" ''
+          unset PATH
+          dontAddDisableDepTrack=1
+          # TODO: https://github.com/NixOS/nix/blob/2.8.0/src/nix-build/nix-build.cc#L506
+          [ -e $stdenv/setup ] && source $stdenv/setup
+          PATH=${lib.escapeShellArg (dirOf shell)}:${lib.makeBinPath [ builder ]}:"$PATH"
+          SHELL=${lib.escapeShellArg shell}
+          BASH=${lib.escapeShellArg shell}
+          set +e
+          [ -n "$PS1" -a -z "$NIX_SHELL_PRESERVE_PROMPT" ] && PS1='\n\[\033[1;32m\][nix-shell:\w]\$\[\033[0m\] '
+          if [ "$(type -t runHook)" = function ]; then
+            runHook shellHook
+          fi
+          unset NIX_ENFORCE_PURITY
+          shopt -u nullglob
+          shopt -s execfail
+          ${optionalString (command != null || run != null) ''
+            ${optionalString (command != null) command}
+            ${optionalString (run != null) run}
+            exit
+          ''}
+        '';
+
+        # https://github.com/NixOS/nix/blob/2.8.0/src/libstore/globals.hh#L464-L465
+        sandboxBuildDir = "/build";
+
+        setupPackage = runCommandNoCC "setup" {
+          # https://github.com/NixOS/nix/blob/ffe155abd36366a870482625543f9bf924a58281/src/libstore/build/local-derivation-goal.cc#L906-L910
+          passwd = ''
+            root:x:0:0:Root user:${sandboxBuildDir}:/noshell
+            nixbld:x:${toString uid}:${toString gid}:Build user:${sandboxBuildDir}:/noshell
+            nobody:x:65534:65534:Nobody:/:/noshell
+          '';
+          group = ''
+            root:x:0:
+            nixbld:!:${toString gid}:
+            nogroup:x:65534:
+          '';
+          passAsFile = [ "passwd" "group" ];
+        } ''
+          # Directories that are later made writable in fakeRootCommands
+          mkdir -p "$out"{/tmp,${storeDir},${sandboxBuildDir}}
+
+          # Directories for internet access and POSIX compat
+          mkdir -p "$out"{/etc/ssl/certs,/bin,/usr/bin}
+          # Allows Nix binaries to find CA certificates for internet access
+          ln -s ${cacert}/etc/ssl/certs/ca-bundle.crt "$out"/etc/ssl/certs/ca-certificates.crt
```

Comment:
Shouldn't it be sufficient to set `SSL_CERT_FILE = "${cacert.out}/etc/ssl/certs/ca-bundle.crt";` in the container config?